src/%IPM/Main.cls

```diff
@@ -2383,11 +2405,178 @@ ClassMethod UpdateLanguageExtensions(pVerbose As %Boolean = 0, pTestOnly As %Boo
+ZPM(pArgs...)
+ If $System.CLS.IsMthd("%IPM.Main", "Shell") { 
+ 	Do ##class(%IPM.Main).Shell(pArgs...) Quit
+ } Else {
+	// TODO: Decide what to do here; definitely needs to support enabling IPM from here as well,
+	// but need to decide what level of customization to provide
+	Quit
+ }
 $$$ENDTAG
 #; Need an extra line down here to avoid the end bracket being on the same line as $$$ENDTAG - a comment is just fine.
 }
 
+ClassMethod EnableIPM(ByRef pCommandInfo)
+{
+	Set quiet = $$$HasModifier(pCommandInfo,"quiet")
+	Set globally = $$$HasModifier(pCommandInfo, "globally")
+	Set localOnly = $$$HasModifier(pCommandInfo, "localOnly")
+	Set version = $$$GetModifier(pCommandInfo, "version")
+	Set namespaces = $$$GetModifier(pCommandInfo, "namespaces")
+	Set useLocal = 1			// var to store the final decision of whether to use local manifest or get from server
+	Set targetVersion = "" 		// var to store the final version of IPM to be installed
+	Kill targetNamespaces 		// multi-dim array to store the final namespaces that need to install IPM
+
+	// Get list of local IPM artifacts under /<instance install dir>/lib/IPM/
+	Set XMLDir = $system.Util.InstallDirectory()
+	For subDirName = "lib", "IPM" {
+		Set XMLDir = ##class(%File).SubDirectoryName(XMLDir, subDirName)
+	}
+  	Set statement = ##class(%SQL.Statement).%New()
+  	Set sc = statement.%PrepareClassQuery("%File", "FileSet")
+  	If $$$ISERR(sc) {
+		Do $SYSTEM.Status.DisplayError(sc)
+		Quit
+	}
+	// Valid IPM installation manifest should be of format: ipm-0.0.1.xml
+	Set resultSet = statement.%Execute(XMLDir, "ipm-*.xml")
+	If ((resultSet.%SQLCODE '= 0) && '(resultSet.%Message [ "does not exist")) {
+		Write !, resultSet.%Message 
+		Quit
+	}
+	Kill ipmLocalArray			// multi-dim array to store local ipm manifests; ipmLocalArray(<version>)=<fileName>
+	Kill menuList				// menu to let user choose which version of local ipm to install (if not in quiet mode and no version is specified)
+	Set counter = 0
+	While resultSet.%Next(.sc) {
+		$$$ThrowOnError(sc)
+		Set fileName = resultSet.%Get("Name")
+		If (resultSet.%Get("Type") = "F") {
+			// get the exact version string from filename
+			Set ipmLocalArray($Piece($Piece(fileName, "ipm-", 2), ".xml")) = fileName
+			Set counter = counter + 1
+			Set menuList(counter) = $Piece($Piece(fileName, "ipm-", 2), ".xml")
+		}
+	}
+	$$$ThrowOnError(sc)
+
+	If (localOnly && (counter = 0)) {
+		$$$ThrowOnError($$$ERROR($$$GeneralError,"No ipm-<version>.xml installer file is found locally."))
+	}
+
+	If (version '= "") {
+		Set targetVersion = version
+		If '$DATA(ipmLocalArray(version)) {
+			If localOnly {
+				$$$ThrowOnError($$$ERROR($$$GeneralError,"No ipm-"_version_".xml installer file is found in "_XMLDir_" locally."))
+			} ElseIf 'quiet {
+				Write !, "No ipm-"_version_".xml installer file is found in "_XMLDir_" locally. Trying to download from the registry."
```

Comment:
Should display this message either way. quiet only means user shouldn't be prompted. 
If you want a modifier for not displaying any messages to the current device, add a `verbose` modifier.